presto-hudi/src/main/java/com/facebook/presto/hudi/HudiPartitionManager.java

```diff
@@ -59,10 +91,198 @@ public List<String> getEffectivePartitions(
         }
 
+        boolean metaTableEnabled = isHudiMetadataTableEnabled(connectorSession);
+
+        return metaTableEnabled ? prunePartitionByMetaDataTable(connectorSession, metaClient, partitionColumns, tupleDomain) :
+                prunePartitionByMetaStore(metastore, metastoreContext, schemaName, tableName, partitionColumns, tupleDomain);
+    }
+
+    private List<String> prunePartitionByMetaDataTable(
+            ConnectorSession connectorSession,
+            HoodieTableMetaClient metaClient,
+            List<Column> partitionColumns,
+            TupleDomain<ColumnHandle> tupleDomain)
+    {
+        // non-partition table
+        if (partitionColumns.isEmpty()) {
+            return ImmutableList.of("");
+        }
+        Configuration conf = metaClient.getHadoopConf();
+        HoodieLocalEngineContext engineContext = new HoodieLocalEngineContext(conf);
+        HoodieMetadataConfig metadataConfig = HoodieMetadataConfig.newBuilder().enable(true).build();
+
+        // Load all the partition path from the basePath
+        List<String> allPartitions = FSUtils.getAllPartitionPaths(engineContext, metadataConfig, metaClient.getBasePathV2().toString());
+
+        // Extract partition columns predicate
+        TupleDomain<String> partitionPredicate = tupleDomain.transform(hudiColumnHandle -> {
+            if (((HudiColumnHandle) hudiColumnHandle).getColumnType() != HudiColumnHandle.ColumnType.PARTITION_KEY) {
+                return null;
+            }
+            return ((HudiColumnHandle) hudiColumnHandle).getName();
+        });
+
+        if (partitionPredicate.isAll()) {
+            return allPartitions;
+        }
+
+        if (partitionPredicate.isNone()) {
+            return ImmutableList.of("");
+        }
+
+        List<HudiColumnHandle> partitionColumnHandles = fromPartitionColumns(partitionColumns);
+
+        List<String> matchedPartitionPaths = prunePartitions(partitionPredicate, partitionColumnHandles, getPartitions(partitionColumns.stream().map(f -> f.getName()).collect(Collectors.toList()), allPartitions));
+        log.info(format("Total partition size is %s, after partition prune size is %s.", allPartitions.size(), matchedPartitionPaths.size()));
+        return matchedPartitionPaths;
+    }
+
+    /**
+     * Returns the partition path key and values as a list of map.
+     * For example:
+     * partition keys: [p1, p2, p3],
+     * partition paths:
+     *      p1=val1/p2=val2/p3=val3  (hive style partition)
+     *      p1=val4/p2=val5/p3=val6  (hive style partition)
+     * return values {p1=val1/p2=val2/p3=val3 -> {p1 -> val1, p2 -> value2, p3 -> value3}}, {p1=val4/p2=val5/p3=val6 -> {p1 -> val4, p2 -> value5, p3 -> value6}}
+     *
+     * @param partitionKey  The partition key list
+     * @param partitionPaths partition path list
+     */
+    public static Map<String, Map<String, String>> getPartitions(List<String> partitionKey, List<String> partitionPaths)
```

Comment:
For a table with non hive-style partitioning, this method returns a key for which parsing fails at `MetastoreUtil#extractPartitionValues`. For example, let us assume the arguments to this method are as follows:
```
partitionKey: [dt]
partitionPaths: [2018-08-31]

return value will be: {2018-08-31 -> {dt -> 2018-08-31}}
```
Note that the key of the returned value is not hive-style partition value. With this map, `extractPartitionValues(entry.getKey())` is called in `HudiSplitManager#getHudiPartitions`. That throws error as below:
```
Caused by: java.lang.IllegalArgumentException: Invalid partition spec: 2018-08-31
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:216)
	at com.facebook.presto.hive.metastore.MetastoreUtil.extractPartitionValues(MetastoreUtil.java:585)
	at com.facebook.presto.hive.metastore.MetastoreUtil.extractPartitionValues(MetastoreUtil.java:554)
	at com.facebook.presto.hudi.HudiSplitManager.lambda$getHudiPartitions$7(HudiSplitManager.java:236)
```

Reply:
One way I could fix this was to add a util method in `HudiPartitionManager` and return early with the partition values if it is non hive style partitioned. Let me know if this makes sense. It would be great if we can add a test for non hive-style partitioned table as well.
```
    static List<String> extractPartitionValues(String partitionName, Optional<List<String>> partitionColumnNames)
    {
        boolean hiveStylePartition = HIVE_PARTITION_NAME_PATTERN.matcher(partitionName).matches();
        if (!hiveStylePartition) {
            if (!partitionColumnNames.isPresent() || partitionColumnNames.get().size() == 1) {
                return ImmutableList.of(partitionName);
            }
            else {
                String[] partitionValues = partitionName.split(Path.SEPARATOR);
                checkArgument(
                        partitionValues.length == partitionColumnNames.get().size(),
                        "Invalid partition spec: {partitionName: %s, partitionColumnNames: %s}", partitionName, partitionColumnNames.get());
                return Arrays.asList(partitionValues);
            }
        }

        return MetastoreUtil.extractPartitionValues(partitionName, partitionColumnNames);
    }

```